lib/client.js

```diff
@@ -597,7 +600,17 @@ MqttClient.prototype._cleanUp = function (forced, done) {
  */
 MqttClient.prototype._sendPacket = function (packet, cb) {
   if (!this.connected) {
-    return this.queue.push({ packet: packet, cb: cb });
+    if (0 === packet.qos && this.qosZeroNotQueued) {
+      if (cb) {
+        cb(new Error('No connection to broker'));
+      } else {
+        this.emit('error', new Error('No connection to broker'));
```

Comment:
I'm not convinced we should error the client for sending a message when not connected.
If it's QoS 0, then you are accepting that the packet can be lost. I think we should just ignore this packet if there is no callback.